src/server/middleware/security/robot.go

```diff
@@ -1,74 +1,76 @@
-// Copyright Project Harbor Authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//    http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package security
 
 import (
-	"net/http"
-	"strings"
-
-	"github.com/goharbor/harbor/src/common"
+	"fmt"
 	"github.com/goharbor/harbor/src/common/security"
 	robotCtx "github.com/goharbor/harbor/src/common/security/robot"
-	ctl_robot "github.com/goharbor/harbor/src/controller/robot"
+	"github.com/goharbor/harbor/src/common/utils"
+	robot_ctl "github.com/goharbor/harbor/src/controller/robot"
+	"github.com/goharbor/harbor/src/lib/config"
 	"github.com/goharbor/harbor/src/lib/log"
-	pkg_token "github.com/goharbor/harbor/src/pkg/token"
-	robot_claim "github.com/goharbor/harbor/src/pkg/token/claims/robot"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/permission/types"
+	"strings"
+	"time"
+
+	"github.com/goharbor/harbor/src/pkg/robot/model"
+	"net/http"
 )
 
 type robot struct{}
 
 func (r *robot) Generate(req *http.Request) security.Context {
 	log := log.G(req.Context())
-	robotName, robotTk, ok := req.BasicAuth()
+	name, secret, ok := req.BasicAuth()
 	if !ok {
 		return nil
 	}
-	if !strings.HasPrefix(robotName, common.RobotPrefix) {
-		return nil
-	}
-	rClaims := &robot_claim.Claim{}
-	defaultOpt := pkg_token.DefaultTokenOptions()
-	if defaultOpt == nil {
-		log.Error("failed to get default token options")
+	if !strings.HasPrefix(name, config.RobotPrefix(req.Context())) {
 		return nil
 	}
-	rtk, err := pkg_token.Parse(defaultOpt, robotTk, rClaims)
+	// The robot name can be used as the unique identifier to locate robot as it contains the project name.
+	robots, err := robot_ctl.Ctl.List(req.Context(), q.New(q.KeyWords{
+		"name": strings.TrimPrefix(name, config.RobotPrefix(req.Context())),
```

Comment:
not sure how the config.RobotPrefix is implemented, but you don't need to call it twice in this func.